### PR TITLE
GH-38164: [MATLAB] Rename `Length` property on `arrow.array.Array` and  `arrow.array.ChunkedArray` to `NumElements`

### LINKED ISSUE #38164: [MATLAB] Rename `Length` property on `arrow.array.Array` and  `arrow.array.ChunkedArray` to `NumElements`
### Describe the enhancement requested

We would like to rename the `Length` property of `arrow.array.Array` to `NumElements` because MATLAB has a function called `length(A)`, which returns the length of the largest array dimension in `A`.  
Because `arrow.array.Array` inherits from `matlab.mixin.Scalar`, it's size is always `1x1`, so the `length()` function always returns `1`. This may confuse users because the `Length` property of `arrow.array.Array` returns the number of elements within the array, which may not be `1`. 

```matlab
>>  array = arrow.array([1 2 3 4 5]);

% The length property returns the number of elements in the array
>> array.Length

ans =

  int64

   5

% The length method returns the length of the largest dimension
>> length(array)

ans =

     1
```

I suspect this inconsistency will confuse users, so we should rename the `Length` property to `NumElements`. 

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

We would like to rename the `Length` property on `arrow.array.Array` (and `arrow.array.ChunkedArray`) to `NumElements` because MATLAB has a function called `length(A)`, which returns the length of the largest array dimension in `A`.  
Because `arrow.array.Array` inherits from `matlab.mixin.Scalar`, it's size is always `1x1`, so the `length()` function always returns `1`. This may confuse users because the `Length` property of `arrow.array.Array` returns the number of elements within the array, which may not be `1`. 

```matlab
>> array = arrow.array([1 2 3 4 5]);

% The length property returns the number of elements in the array
>> array.Length

ans =

  int64

   5

% The length method returns the length of the largest dimension
>> length(array)

ans =

     1
```

I suspect this inconsistency will confuse users, so we should rename the `Length` property to `NumElements`. 

### What changes are included in this PR?

1. Renamed `Length` property on `arrow.array.Array` to `NumElements`
2. Renamed `Length` property on `arrow.array.ChunkedArray` to `NumElements`

### Are these changes tested?

Yes. I modified the existing `Length` test cases to now test the `NumElements` property.

### Are there any user-facing changes?

Yes. Please note this is a breaking change because `Length` is no longer a property on either `arrow.array.Array` or `arrow.array.ChunkedArray`. However, we have not yet cut an initial release of the MATLAB interface, so we don't expect this change should affect too many users at this point. 

* Closes: #38164

### CODE REVIEW

File: matlab/src/matlab/+arrow/+tabular/+internal/validateArrayLengths.m [hunk 1]
```diff
@@ -24,14 +24,14 @@ function validateArrayLengths(arrowArrays)
...
+    expectedNumElements = arrowArrays{1}.NumElements;
 
     for ii = 2:numel(arrowArrays)
-        if arrowArrays{ii}.Length ~= expectedLength
+        if arrowArrays{ii}.NumElements ~= expectedNumElements
             errid = "arrow:tabular:UnequalArrayLengths";
-            msg = compose("Expected all arrays to have a length of %d," + ...
-                " but the array at position %d has a length of %d.", ...
-                expectedLength, ii, arrowArrays{ii}.Length);
+            msg = compose("Expected all arrays to %d elements," + ...
```
Comment (kevingurney) [inline matlab/src/matlab/+arrow/+tabular/+internal/validateArrayLengths.m:32]: to *have* %d elements

File: matlab/test/arrow/array/tChunkedArray.m [hunk 2]
```diff
@@ -413,20 +413,20 @@ function ToMATLABTimestampType(testCase, TimeZone)
             a2 = arrow.array(dates(6:5));
             a3 = arrow.array(dates(6:end));
 
-            % ChunkedArray with three chunks and nonzero length
+            % ChunkedArray with three chunks and nonzero at least one element
```
Comment (kevingurney) [inline matlab/test/arrow/array/tChunkedArray.m:416]: nonzero at least on element -> at least one element

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thank you for making this change! This is a nice usability improvement and should hopefully reduce unnecessary confusion.

**Review by kou — APPROVED** [review]
+1

File: matlab/src/matlab/+arrow/+tabular/+internal/validateArrayLengths.m (hunk 1 above)
Comment (sgilmore10) [reply matlab/src/matlab/+arrow/+tabular/+internal/validateArrayLengths.m:32]: ah good catch!

**Review by kevingurney — APPROVED** [review]

**Pull request merged**